### PR TITLE
"added config / subclasses"

### PR DESCRIPTION
**Pull request by MarinaMuttik**

### CODE REVIEW

**Comment by dansteele** [comment]
So can you describe what the issue is here? 

You can use this format to use syntax highlighting in your comments :)

```ruby
class Hello
  def hi
    'asd'
  end
end
```

screenshot 2018-10-17 at 17 57 01

**Comment by MarinaMuttik** [comment]
> So can you describe what the issue is here?

So according to <a href="http://recipes.sinatrarb.com/p/middleware/rack_auth_basic_and_digest"> this</a>, using the code I have across .rb and .ru upon running rackup -p 4567 it should display the page as normal, and in /workhistory then require authentication there. But, instead on running rackup all that I get is an error after it reads the require file, goes there and when it reaches the first line of .rb I get: 
```ruby
require 'sinatra/base'
```

> /webcv.rb:1:in `require': cannot load such file -- sinatra/base (LoadError)

And I'm unsure why as the version in master has the same format and is capable of requiring both sinatra & sinatra/base when run directly. Hope that makes some sense..!

**Comment by dansteele** [comment]
Can you push the latest you got :)

File: config.ru [hunk 1]
```diff
@@ -0,0 +1,6 @@
+require File.expand_path '../webcv.rb', __FILE__
+
+run Rack::URLMap.new({
+  "/" => Public,
+  "/workhistory" => Protected
+  })
```
Comment (dansteele) [inline config.ru:7]: Stay on top of your indentation :)

File: webcv.rb [hunk 2]
```diff
@@ -1,10 +1,20 @@
...
+end
+
+class Protected < Sinatra::Base
+
+  use Rack::Auth::Basic, "Authentication" do |username, password|
+    username == "marina" && password == "workhis1"
+  end
 
-get '/workhistory' do
+get '' do
```
Comment (dansteele) [inline webcv.rb]: indentation :)

File: webcv.rb [hunk 3]
```diff
@@ -1,10 +1,20 @@
-require 'sinatra'
+require 'sinatra/base'
 require 'slim'
 
+class Public < Sinatra::Base
```
Comment (dansteele) [inline webcv.rb]: The general convention is to have one class per file

**Review by dansteele — CHANGES_REQUESTED** [review]
I think it might be worth spending some [more] time working with classes, as in just building your own 

Maybe spend some more time on problem 22 to see how you could refactor it in to more distinct methods, using attr_accessors and instance variables. At the moment everything is in one `perform` method and so you're not making much use of classes.

The reason (I think) that this isn't behaving that way you're expecting it to is because you've got some random classes here, but nothing's 'happening' with them, they just kind of _exist_. Which to me says it's probably a good idea to get more comfortable with classes! :)

I have a 'model answer' for problem 22 which I wrote, which I can share with you after you've spent some more time on it, if you'd like :)

**Pull request merged**